src/common/entity/compute_attribute_display.ts

```diff
@@ -31,7 +39,35 @@ export const computeAttributeValueDisplay = (
 
   // Number value, return formatted number
   if (typeof attributeValue === "number") {
-    return formatNumber(attributeValue, locale);
+    const formattedValue = formatNumber(attributeValue, locale);
+
+    const domain = computeStateDomain(stateObj);
+
+    let unit =
+      DOMAIN_ATTRIBUTES_UNITS[domain]?.[attribute] ??
+      (GLOBAL_ATTRIBUTES_UNITS[attribute] as string | undefined);
+
+    if (domain === "weather") {
+      unit = getWeatherUnit(config, stateObj as WeatherEntity, attribute);
+    }
+
+    if (unit === "%") {
+      return `${formattedValue}${blankBeforePercent(locale)}${unit}`;
+    }
+
+    if (unit === "°") {
+      return `${formattedValue}${unit}`;
+    }
+
+    if (unit) {
+      return `${formattedValue} ${unit}`;
+    }
+
+    if (TEMPERATURE_ATTRIBUTES.has(attribute)) {
+      return `${formattedValue} ${config.unit_system.temperature}`;
```

Comment:
Is this correct? As we have entity settings for the temperature unit too? Should this use the entity UOM if it is als temperature? 😅 Or is the unit of attribute values not changeable yet?

Reply:
We can not change it. We used everywhere `config.unit_system.temperature` before this PR.
There is a special case for weather entity but for climate and water_heater it's not editable.
Do you have any other domain that have temperature attribute?

Reply:
> Do you have any other domain that have temperature attribute?
I don't know, I think it is fine, people should use sensors for all other things anyway. I can imagine for climate and weather we will introduce a setting for units too, then I think we should check the entity reg